src/imagery/i.eodag/i.eodag.py

```diff
@@ -139,54 +151,91 @@ def get_bb(vector=None):
     }
 
 
+def get_aoi(vector=None):
+    """Get the AOI for querying"""
+    # Handle empty AOI
+    if not vector:
+        return get_bb()
+
+    # TODO: Read Vector WKB
+
+
 def download_by_id(query_id: str):
-    gs.verbose(
-        _(
-            "Searching for product ending with: {}".format(
-                query_id[-min(len(query_id), 8) :]
-            )
-        )
-    )
-    product, count = dag.search(id=query_id)
+    gs.message(_("Attempting to download product: {}".format(query_id)))
+    product, count = dag.search(id=query_id, provider=options["provider"])
     if count != 1:
-        raise ParameterError("Product couldn't be uniquely identified")
+        raise ParameterError("Product couldn't be uniquely identified.")
     if not product[0].properties["id"].startswith(query_id):
-        raise ParameterError("Product wasn't found")
-    gs.verbose(
-        _("Poduct ending with: {} is found.".format(query_id[-min(len(query_id), 8) :]))
-    )
-    gs.verbose(_("Downloading..."))
+        raise ParameterError("Product wasn't found.")
+    gs.verbose(_("Poduct {} is found.".format(query_id)))
+    gs.verbose(_("Downloading {}".format(query_id)))
     dag.download(product[0])
 
 
+def ids_from_file_txt(ids_file_txt):
+    ids_set = set()
+    with open(ids_file_txt, "r") as ids_stream:
+        line_index = 0
+        lines = ids_stream.read().split("\n")
+        for line in lines:
+            line_index += 1
+            if not line:
```

Comment:
Here you can use enumerate...
```suggestion
        for line_index, enumerate(line in lines):
            if not line:
```